Create Documentation for the ComInterfaceGenerator (#36833)
dotnet/docs: docs/standard/native-interop/cominterop.md

The Component Object Model (COM) lets an object expose its functionality to other components and to host applications on Windows platforms. To help enable users to interoperate with their existing code bases, .NET Framework has always provided strong support for interoperating with COM libraries. In .NET Core 3.0, a large portion of this support has been added to .NET Core on Windows. The documentation here explains how the common COM interop technologies work and how you can utilize them to interoperate with your existing COM libraries.

## Built-in and source-generated COM interop

> COM interop functionality can be achieved through a built-in system in the .NET runtime or through implementing the [ComWrappers API](./tutorial-comwrappers.md) (introduced in .NET 6). In .NET 8, the [COM source generator](./comwrappers-source-generation.md) can be used to automatically implement the `ComWrappers` API for `IUnknown`-based interfaces.

Reviewer: ```suggestion
COM interop functionality can be achieved through a built-in system in the .NET runtime or through implementing the [ComWrappers API](./tutorial-comwrappers.md) (introduced in .NET 6). Starting in .NET 8, you can use the [COM source generator](./comwrappers-source-generation.md) to automatically implement the `ComWrappers` API for `IUnknown`-based interfaces.
```

> COM interop functionality can be achieved though a system built in to the .NET runtime (introduced in .NET Core 3.0), or through implementing the [ComWrappers API](./tutorial-comwrappers.md) (introduced in .NET 6). In .NET 8, the [COM source generator](./comwrappers-source-generation.md) can be used to automatically implement the ComWrappers API for IUnknown based interfaces.

Reviewer 2: ```suggestion
COM interop functionality can be achieved though a system built-in to the .NET runtime (introduced in .NET Core 3.0), or through implementing the [ComWrappers API](./tutorial-comwrappers.md) (introduced in .NET 6). In .NET 8, the [COM source generator](./comwrappers-source-generation.md) can be used to automatically implement the ComWrappers API for `IUnknown` based interfaces.
```

> COM interop functionality can be achieved though a system built-in to the .NET runtime (introduced in .NET Core 3.0), or through implementing the [ComWrappers API](./tutorial-comwrappers.md) (introduced in .NET 6). In .NET 8, the [COM source generator](./comwrappers-source-generation.md) can be used to automatically implement the ComWrappers API for `IUnknown` based interfaces.

Reviewer 3: ```suggestion
COM interop functionality can be achieved through a built-in system in the .NET runtime (introduced in .NET Core 3.0) or through implementing the [ComWrappers API](./tutorial-comwrappers.md) (introduced in .NET 6). In .NET 8, the [COM source generator](./comwrappers-source-generation.md) can be used to automatically implement the ComWrappers API for `IUnknown`-based interfaces.
```

> COM interop functionality can be achieved through a built-in system in the .NET runtime (introduced in .NET Core 3.0) or through implementing the [ComWrappers API](./tutorial-comwrappers.md) (introduced in .NET 6). In .NET 8, the [COM source generator](./comwrappers-source-generation.md) can be used to automatically implement the ComWrappers API for `IUnknown`-based interfaces.

Reviewer 2: ```suggestion
COM interop functionality can be achieved through a built-in system in the .NET runtime or through implementing the [ComWrappers API](./tutorial-comwrappers.md) (introduced in .NET 6). In .NET 8, the [COM source generator](./comwrappers-source-generation.md) can be used to automatically implement the `ComWrappers` API for `IUnknown`-based interfaces.
```

I think this page is for both .NET Framework and .NET 5+. The statement about .NET Core 3.0 is a bit confusing as it relates to .NET Framework so I removed it.
